src/Library/WebServer.cs

```diff
@@ -113,6 +115,8 @@ internal void ListenerCallbackAsync(IAsyncResult result)
                         ClusterIdPayloadObject payloadObject = (ClusterIdPayloadObject)serializer.ReadObject(request.InputStream);
                         Diagnostics.LogInfo(FormattableString.Invariant($"received payload with cluster id : {payloadObject.clusterId}"));
                         response.StatusCode = (int)HttpStatusCode.Accepted;
+
+                        this.publisher.UpdateClusterId(payloadObject.clusterId);
```

Comment:
Can payloadObject be null?

Reply:
if it is , or there are issues at deserialization we will catch it below, set the response as bad request and return it to the caller